src/components/table/table.tsx

```diff
@@ -328,7 +387,24 @@ export class Table {
     }
 
     private getColumnDefinitions(): Tabulator.ColumnDefinition[] {
-        return this.columns.map(this.columnFactory.create);
+        const columnDefinitions = this.columns.map(this.columnFactory.create);
+
+        if (this.selectable) {
+            return [
+                this.getRowSelectorColumnDefinition(),
+                ...columnDefinitions,
+            ];
+        }
+
+        return columnDefinitions;
+    }
+    private getRowSelectorColumnDefinition(): TabulatorTable.ColumnDefinition {
+        return {
+            title: '',
+            formatter: 'rowSelection',
```

Comment:
@anderssonjohan asked:
> built-in Tabulator formatter rowSelection captures the checkbox click event and prevents the expected shift-click extension of the selection when clicking the checkboxes. Shift-click to extend only works when clicking outside the checkbox! This should be replaced by our own "rowSelection column", which also opens up the possibility to use a limel-checkbox for improved styling

The default tabulator behavior of shift+click to create selections is really bad. For example, you cannot manually select row 1, then 2, and then row 8 and shift+click to select from row 8 to row 20. This interaction will result in selecting everything from row 1 to 20. This sucks.

But do we really want to correct them? I'd say it will become a deep rabbit hole, and even if we fix it, upgrading Tabulator to next versions will kill us.

> which also opens up the possibility to use a limel-checkbox for improved styling

Don't worry for styling, and we don't _have to_ use limel-checkbox if it's hard. We can always make their checkbox look like ours.

Reply:
Look at the correct shift+click behavior 

https://user-images.githubusercontent.com/35954987/149308834-543c4b44-1a36-4655-a5fe-2fac2b4ecf37.mov

Reply:
> @anderssonjohan asked:
> 
> > built-in Tabulator formatter rowSelection captures the checkbox click event and prevents the expected shift-click extension of the selection when clicking the checkboxes. Shift-click to extend only works when clicking outside the checkbox! This should be replaced by our own "rowSelection column", which also opens up the possibility to use a limel-checkbox for improved styling
> 
> The default tabulator behavior of shift+click to create selections is really bad. For example, you cannot manually select row 1, then 2, and then row 8 and shift+click to select from row 8 to row 20. This interaction will result in selecting everything from row 1 to 20. This sucks.
> 
> But do we really want to correct them? I'd say it will become a deep rabbit hole, and even if we fix it, upgrading Tabulator to next versions will kill us.

Nah, it won't be that bad. The built-in rowSelection formatter does actually keep track of the last clicked row, but the resulting behavior is like you wrote; it selects everything between the first selected row and the one you shift-click.

I'm probably going to make a Tabulator add-on similar to the Knockout custom binding I did (9 years ago! 🤭) in Lime Field: 
https://github.com/Lundalogik/field-web-commons/commit/90eeaf21c0762909868d8d197f2359dcc3602b8b (internal link)

It provides the behavior you describe and we want:

https://user-images.githubusercontent.com/435885/149316074-997b2ff4-cbe6-4312-855e-d192f48ace8b.mov

Reply:
Well. I'm not a developer. So do what's right 😊

Reply:
Just don't do _round_ checkboxes! 😝

Reply:
> Just don't do _round_ checkboxes! 😝

You're banging on open doors, I'm not the designer of this ;)